plugins/vppv2/puntplugin/vppcalls/punt_vppcalls.go

```diff
@@ -0,0 +1,241 @@
+	for i, c := range pathName {
+		pathByte[i] = c
+	}
+
+	req := &api.PuntSocketRegister{
+		HeaderVersion: 1,
+		IsIP4:         boolToUint(isIPv4),
+		L4Protocol:    resolveL4Proto(punt.L4Protocol),
+		L4Port:        uint16(punt.Port),
+		Pathname:      pathByte,
+	}
+	reply := &api.PuntSocketRegisterReply{}
+
+	if err := h.callsChannel.SendRequest(req).ReceiveReply(reply); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h *PuntVppHandler) unregisterPuntWithSocketIPv4(punt *punt.ToHost) error {
+	return h.unregisterPuntWithSocket(punt, true)
+}
+
+func (h *PuntVppHandler) unregisterPuntWithSocketIPv6(punt *punt.ToHost) error {
+	return h.unregisterPuntWithSocket(punt, false)
+}
+
+func (h *PuntVppHandler) unregisterPuntWithSocket(punt *punt.ToHost, isIPv4 bool) error {
+	req := &api.PuntSocketDeregister{
+		IsIP4:      boolToUint(isIPv4),
+		L4Protocol: resolveL4Proto(punt.L4Protocol),
+		L4Port:     uint16(punt.Port),
+	}
+	reply := &api.PuntSocketDeregisterReply{}
+
+	if err := h.callsChannel.SendRequest(req).ReceiveReply(reply); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h *PuntVppHandler) handlePuntRedirectIPv4(punt *punt.IpRedirect, isAdd bool) error {
+	return h.handlePuntRedirect(punt, true, isAdd)
+}
+
+func (h *PuntVppHandler) handlePuntRedirectIPv6(punt *punt.IpRedirect, isAdd bool) error {
+	return h.handlePuntRedirect(punt, false, isAdd)
+}
+
+func (h *PuntVppHandler) handlePuntRedirect(punt *punt.IpRedirect, isIPv4, isAdd bool) error {
+	// rx interface
+	var rxIfIdx uint32
+	if punt.RxInterface == "" {
+		rxIfIdx = ^uint32(0)
+	} else {
+		rxMetadata, exists := h.ifIndexes.LookupByName(punt.RxInterface)
+		if !exists {
+			return errors.Errorf("index not found for interface %s", punt.RxInterface)
+		}
+		rxIfIdx = rxMetadata.SwIfIndex
+	}
+
+	// tx interface
+	txMetadata, exists := h.ifIndexes.LookupByName(punt.TxInterface)
+	if !exists {
+		return errors.Errorf("index not found for interface %s", punt.TxInterface)
+	}
+
+	// next hop address
+	var nextHop []byte
+	if isIPv4 {
+		nextHop = net.ParseIP(punt.NextHop).To4()
+	} else {
+		nextHop = net.ParseIP(punt.NextHop).To16()
+	}
+
+	req := &ipApi.IPPuntRedirect{
+		IsAdd:       boolToUint(isAdd),
+		IsIP6:       boolToUint(!isIPv4),
+		RxSwIfIndex: rxIfIdx,
+		TxSwIfIndex: txMetadata.SwIfIndex,
+		Nh:          nextHop,
+	}
+	reply := &ipApi.IPPuntRedirectReply{}
+	if err := h.callsChannel.SendRequest(req).ReceiveReply(reply); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resolveL3Proto(protocol punt.L3Protocol) uint8 {
+	switch protocol {
+	case punt.L3Protocol_IPv4:
+		return 4 // binary API IPv4 representation
+	case punt.L3Protocol_IPv6:
+		return 6 // binary API IPv6 representation
+	default:
+		return ^uint8(0) // binary API representation for both protocols
+	}
+}
+
+func resolveL4Proto(protocol punt.ToHost_L4Protocol) uint8 {
+	switch protocol {
+	case punt.ToHost_TCP:
+		return 6 // TCP (IANA)
+	default:
```

Comment:
Do not use _default_ case as the fallback, even if there is only single other value to prevent unintended behavior when new values are added in the future.